packages/dds/tree/src/test/simple-tree/schemaFactoryRecursive.spec.ts

```diff
@@ -166,6 +174,79 @@ describe("SchemaFactory Recursive methods", () => {
 			);
 		});
 
+		it("object with required recursive field", () => {
+			class ObjectRecursive extends sf.objectRecursive("Object", {
+				x: sf.requiredRecursive([() => ObjectRecursive, sf.number]),
+			}) {}
+			{
+				type _check = ValidateRecursiveSchema<typeof ObjectRecursive>;
+			}
+
+			type XSchema = typeof ObjectRecursive.info.x;
+			type Field2 = XSchema extends FieldSchema<infer Kind, infer Types>
+				? ApplyKind<TreeNodeFromImplicitAllowedTypes<Types>, Kind, false>
+				: "zzz";
+			type XTypes = XSchema extends FieldSchemaUnsafe<infer Kind, infer Types> ? Types : "Q";
```

Comment:
Using string constants for the error cases in conditionals is a great idea for making better errors. I suspect it might not be intuitive to people reading this code later though. I think we should use descriptive strings for even better errors anbd more self documenting code:

```suggestion
				: "Not A FieldSchema";
			type XTypes = XSchema extends FieldSchemaUnsafe<infer Kind, infer Types> ? Types : "Not A FieldSchemaUnsafe";
```